homeassistant/components/powerwall/sensor.py

```diff
@@ -132,7 +132,7 @@ def _get_meter_average_voltage(meter: MeterResponse) -> float:
         state_class=SensorStateClass.MEASUREMENT,
         device_class=SensorDeviceClass.VOLTAGE,
         native_unit_of_measurement=UnitOfElectricPotential.VOLT,
-        value_fn=lambda battery_data: round(battery_data.v_out, 1),
+        value_fn=lambda battery_data: round(battery_data.v_out or 0.0, 1),
```

Comment:
This looks like it should be unavailable if the value is `None`?

Reply:
I'm just trying to avoid `TypeError: type NoneType doesn't define __round__ method` here. Is there a better approach?

Reply:
Change the code that consumes this so it sets `available` to `False` if its None and don't run the `value_fn` if its not available. You might need to create an `available_fn`

Reply:
I follow you, but it sounds like we're outside the boundaries of my Python competency.  There's no place else in the component where `v_out` appears, so I don't know where "the code that consumes this" lives.  Do you want to make the changes here, or should I close this PR and let you handle it?

Reply:
You just have to look at the purpose of `value_fn` in this file and duplicate the same design but for the [available](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L824) method.